src/packages/utils/src/things/logger.ts

```diff
@@ -0,0 +1,70 @@
+import { createWriteStream } from "fs";
+import { format } from "util";
+export type LogFunc = (message?: any, ...optionalParams: any[]) => void;
+
+export type Logger = {
+  log: LogFunc;
+};
+
+export type InternalLogger = Logger & {
+  close: () => Promise<void>;
+};
+
+type LoggerConfig = {
+  baseLogger: Logger;
+  file?: number;
+};
+
+export function createLogger(config: LoggerConfig): InternalLogger {
+  const baseLog = (...params: any[]) => config.baseLogger.log(...params);
+
+  if ("file" in config && config.file !== undefined) {
+    if (typeof config.file !== "number") {
+      throw new Error(
+        `'config.file' was not correctly normalized to a file descriptor. This should not happen. Value: ${
+          config.file
+        } of type ${typeof config.file}`
+      );
+    }
+    const fd = config.file;
+
+    const diskLogFormatter = (message: any) => {
```

Comment:
Is `message` always a string? And if so, we don't need the `toString()` call below.